src/develop/masks/gradient.c

```diff
@@ -613,15 +613,23 @@ static int _gradient_get_points(dt_develop_t *dev, float x, float y, float rotat
   (*points)[4] = x2;
   (*points)[5] = y2;
 
-  *points_count = 3;
+  const int nthreads = omp_get_max_threads();
+  uint32_t *pts_count = calloc(nthreads, sizeof(uint32_t));
```

Comment:
This causes false sharing because 16 threads store their counts in the same cache line.  Better to use `dt_alloc_perthread` and later `dt_get_perthread`.

Reply:
But I need to get access to the results in `pts` after the omp for to fill points buffer with the points calculated by the for loop and I don't know before hand the number of points generated by each iteration. Any solution for this ?

Reply:
```
size_t padded_size;
uint32_t *pts_count = dt_alloc_perthread(nthreads, sizeof(uint32_t), &padded_size);
...
#pragma omp parallel for
for() {
   uint32_t *pcount = dt_get_perthread(pts_count, padded_size);
   *pcount++;
}
...
for(thread = 0; thread < nthreads; thread++) {
   count = *dt_get_bythread(pts_count, padded_size, thread);
   ...
}
```

Reply:
I see, thanks will test that.

Reply:
Fixed now, thanks! If there is no more comment I'll merge.